pkg/systemd/systemd.go

```diff
@@ -64,15 +65,21 @@ func ReadConfFile() (spec *SriovConfig, err error) {
 	return spec, err
 }
 
-func WriteConfFile(newState *sriovnetworkv1.SriovNetworkNodeState) (bool, error) {
+func WriteConfFile(newState *sriovnetworkv1.SriovNetworkNodeState, netPoolConfig *sriovnetworkv1.SriovNetworkPoolConfig) (bool, error) {
 	newFile := false
 	// remove the device plugin revision as we don't need it here
 	newState.Spec.DpConfigVersion = ""
 
+	// shared mode is a default on OS
+	rdmaMode := consts.RdmaSubsystemModeShared
```

Comment:
I think we should try to query/change mode only in case if rdmaMode parameter is explicitly set in the poolConfig, to provide a safer behavior for ENVs which doesn't use RDMA.